messaging/src/main/java/org/axonframework/eventhandling/configuration/DefaultEventHandlingComponentBuilder.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventhandling.configuration;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.eventhandling.EventHandler;
+import org.axonframework.eventhandling.EventHandlingComponent;
+import org.axonframework.eventhandling.SequenceOverridingEventHandlingComponent;
+import org.axonframework.eventhandling.async.SequencingPolicy;
+import org.axonframework.messaging.QualifiedName;
+
+import java.util.Objects;
+import java.util.Set;
+import java.util.function.UnaryOperator;
+
+/**
+ * Default implementation of the {@link EventHandlingComponentBuilder} providing a fluent API for configuring
+ * {@link EventHandlingComponent} instances with event handlers, sequencing policies, and decorators.
+ * <p>
+ * This builder follows the builder pattern with type-safe phases to ensure proper configuration order:
+ * <ol>
+ * <li>Optional sequencing policy configuration</li>
+ * <li>Required event handler registration (at least one)</li>
+ * <li>Optional additional event handlers</li>
+ * <li>Optional component decoration</li>
+ * <li>Final component building</li>
+ * </ol>
+ * <p>
+ * The builder requires an existing {@link EventHandlingComponent} as its base component.
+ * When a {@link org.axonframework.eventhandling.async.SequencingPolicy} is applied, the component is wrapped 
+ * in a {@link SequenceOverridingEventHandlingComponent} to provide custom event sequencing behavior.
+ * <p>
+ * This builder is typically used internally by the framework's configuration system. Users should access
+ * it through {@link org.axonframework.configuration.MessagingConfigurer} or similar configuration APIs
+ * rather than instantiating it directly.
+ * <p>
+ * Example usage:
+ * <pre>{@code
+ * EventHandlingComponent baseComponent = new SimpleEventHandlingComponent();
+ * EventHandlingComponent component = new DefaultEventHandlingComponentBuilder(baseComponent)
+ *     .sequencingPolicy(event -> Optional.of(event.aggregateIdentifier()))
+ *     .handles(new QualifiedName("example", "OrderPlaced"), orderEventHandler)
+ *     .handles(new QualifiedName("example", "OrderShipped"), shippingEventHandler)
+ *     .decorated(component -> new DecoratingEventHandlingComponent(component))
+ *     .build();
+ * }</pre>
+ *
+ * @author Mateusz Nowak
+ * @since 5.0.0
+ */
+public class DefaultEventHandlingComponentBuilder
```

Comment:
After I've started to work on annotation-based configuration for the EventHandlingComponent, I think this class will eventually become a Module, a declarative one and the will be "annotated" as well for EventHandlingComponentModle implementation, similar to what we have in EventSourcedEntityModule. 
I'm just not sure if it will be a module on its own or some "Configurer" with 2 variations.